backend/rent.py

```
@@ -0,0 +1,158 @@
+            cursor.execute(insert_query,values)
+            db.commit()
+        except mysql.connector.Error as e:
+            if request.is_json:
+                    return jsonify({"message": "Database error", "error": str(e)}), 500
+            else:
+                flash("Something went wrong")
+                return redirect(url_for('signup'))
+        finally:
+            cursor.close()
+            db.close()
+            return redirect(url_for('login'))  
```

Ruff (0.12.2)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)